TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -0,0 +1,23 @@
+﻿using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Parser;
+using TypeCobol.Compiler.Parser.Generated;
+
+namespace TypeCobol.Compiler.Sql.CodeElements
+{
+    class SqlCodeElementCheckers
+    {
+        public static void OnSavepointStatementChecker(CodeElement codeElement, CodeElementsParser.SavepointStatementContext context)
+        {
+            if (context.savepoint_name == null) return;
+            var savePointName = context.savepoint_name.Text;
+            if (savePointName.Length == 0 || savePointName.Length <= 3) return;
+            var text = savePointName.Substring(0, 3);
+            if (text.ToUpper() == "SYS")
+            {
+                DiagnosticUtils.AddError(codeElement, "Invalid savepoint-name, it must not begin with 'SYS'.",
+                    context);
+            }
```

Comment:
This can be simplified. In such checker methods we don't use the ANTLR context but the constructed code element. The context is mainly used for error positioning.
Also you have to use [string.StartsWith](https://docs.microsoft.com/fr-fr/dotnet/api/system.string.startswith?view=netframework-4.7.2#system-string-startswith(system-string-system-stringcomparison)) method with case insensitive option. This is faster than allocating a new string in `ToUpper`.